### ISSUE TITLE
SuperMatter engine, on selection, starts with none of it's pipenets properly built.

### ISSUE THREAD

**Issue by Hawk-v3** [Repede/YawnWiderStation#168]
#### Brief description of the issue
The Blue, Red, Black and both Cooling Array Lines do NOT initialise correctly, and the engine doesn't function.

#### What you expected to happen
Standard functioning.

#### What actually happened
Had to wrench and rewrench every pipenet involved. The black lines still don't work or register.

#### Steps to reproduce
- Enginespawn the SM with the console.
- Try to set engine up.

#### Code Revision
- Server revision:  - 2018-07-03
be815959c410dbf7d06021be9984d3e24d570b7b
 
#### Anything else you may wish to add:
- Probably caused by the Engine Switching proc, given that the SM layout hasn't changed, nor has any pipe code, I THINK?

Last verified working about... A week ago, IIRC?

**Comment by Hawk-v3** [comment]
@gozulio  I suspect that the pipenets are not being rebuilt once the enginespawner applies the submap.

**Comment by gozulio** [comment]
@Hawk-v3 I've gone through engine setup with the Supermatter 3 times, and all three times the Atmospherics equipment initialized without issue. This is the log i get when the engine is selected, and this is the log the server should get as well.
`Chose Engine Map: Supermatter Engine
Annihilating objects in engine loading location.
Annihilated 670 objects.
Annihilated 0 Living Mobs
Initializing newly created atom(s) in submap.
Initializing atmos pipenets and machinery in submap.
Rebuilding powernets due to submap creation.
Submap initializations finished.`

If there are any further developments on this please try to include the admin log that appears and more detailed accounts of the person(s) setting the engine up. Unfortunately there is not sufficient information to go on in order to investigate much further. I will however poke the person who brought this to our attention for a response.

**Comment by DeepIndigo** [comment]
When I first saw the bug, I selected the supermatter, and then immediately went to go set it up after the common channel announcement (within approximately 5 or 6 seconds of the announce hitting the common channel). The second time, I waited approximately 20 seconds. In both cases the vent connected to the red turbine pipe started "on hold" instead of open.  That's really all I have. I'll keep a close watch every time I set up the engine to see if I can't finds any more indicators.
